lib/adal.js

```diff
@@ -475,11 +475,11 @@ var AuthenticationContext = (function () {
         var frameCheck = frameName;
         setTimeout(function () {
             var frameHandle = self._addAdalFrame(frameCheck);
-            if (frameHandle.src === '' || frameHandle.src === 'about:blank') {
+            if (!frameHandle.src || frameHandle.src === 'about:blank') {
                 frameHandle.src = urlNavigate;
                 self._loadFrame(urlNavigate, frameCheck);
             }
-        }, 500);
+        }, 0);
```

Comment:
Kirk: "The failure in IE is not a race. It is a reentrancy issue. Just letting the stack unwind and then immediately setting the .src works."